File: Language/C#/BubbleSort.cs
/*
 *	Autor: Lockna
 *	This program should provide a simple buble sort implementation
 *	At the end it prints out all numbers, 
 *
*/

using System;

namespace BubbleSort {
    class Bubble {

        static void Main(string[] args) 
        {
            int[] arr = { 78, 55, 45, 98, 13};
            int temp;

            Console.WriteLine("Bubble Sort");
            Console.Write("Initial array is: ");


            // Prints the unchanged array
            foreach (int p in arr)
              Console.Write(p + " ")

            for (int j = 0; j <= arr.Length - 2; j++) 
            {
         
		            for (int i = 0; i <= arr.Length - 2; i++) 
		            {
                    if (arr[i] > arr[i + 1]) 
		                {
                        // If a number is bigger than the successor they get swaped.
                        temp = arr[i + 1];
                        arr[i + 1] = arr[i];
                        arr[i] = temp;
                    }
                }
            }

        Console.WriteLine("Sorted:");
        
        // Prints the sorted key
        foreach (int p in arr)
            Console.Write(p + " ");
         
        // Holds the console window on windows open, since it closes immediately after program end.
        Console.ReadKey();

      }
   }
}

File: Language/C++/BubbleSort.cpp
//program to demonstrate Bubble Sort in c++
//The array to be sorted is same as the array used in the program to demonstrate Selection Sort
#include<iostream>

using namespace std;

void swap(int *xp, int *yp)
{
	int temp = *xp;
	*xp = *yp;
	*yp = temp;
}

void bubblesort(int arr[], int n)
{
	//array is sorted from the end
	for (int t = 0; t < n-1; t++)
	{
		//the maximum element in the array from 0 to n-t-1 is moved to (n-t-1)th position
		for (int i = 0; i < n - t - 1; i++)
		{
			if(arr[i]>arr[i+1])
				swap(&arr[i], &arr[i+1]);
		}
	}
}

void printArray(int arr[], int size)
{
	int i;
	for (i = 0; i < size; i++)
		cout << arr[i] << " ";
	cout << endl;
}

int main()
{
	int arr[] = { 64, 25, 12, 22, 11 };
	int n = sizeof(arr) / sizeof(arr[0]);

	bubblesort(arr, n);

	cout << "Sorted array: \n";
	printArray(arr, n);

	return 0;
}